web/controllers/utils.ts

```diff
@@ -0,0 +1,50 @@
+import {Division, Expertise} from "../../common/entity/Mentor";
+import {getLogger} from "log4js";
+import {ApiSubject} from "./format";
+import {checkSubject} from "./userController/format";
+
+const logger = getLogger();
+
+export function checkSubjects(subjects: ApiSubject[]) {
+    if (subjects.length > 0) {
+        for (let i = 0; i < subjects.length; i++) {
+            if (!checkSubject(subjects[i].name)) {
+                logger.warn("Subjects contain invalid subject " + subjects[i].name);
+                return null;
+            }
+        }
+    }
+    return JSON.stringify(subjects);
+}
+
+export function checkDivisions(divisions: string[]) {
+    let result: Division[] = [];
+    if (divisions.length > 0) {
+        for (let i = 0; i < divisions.length; i++) {
+            if (divisions[i].toUpperCase() in Division) {
+                result.push(Division[divisions[i].toUpperCase()]);
+            } else {
+                logger.warn("Division '" + divisions[i] + "' is not a correct division");
+                return null;
+            }
+        }
+    }
+    return result;
+}
+
+export function checkExpertises(expertises: string[]) {
+    let result: Expertise[] = [];
+    if (expertises.length > 0) {
+        const expertiseValues: string[] = Object.keys(Expertise).map(key => Expertise[key]).filter(k => !(parseInt(k) >= 0));
```

Comment:
Yes, the filter filters values which are no strings. Actually it's not necessary to filter them out, because we are only using string values. 
In the case to make the code easier to read and understand its better to remove the filter.